### ISSUE TITLE
null returned as project_name from listjobs 'list_em_all' version

### ISSUE THREAD

**Issue by khalistoo** [scrapy/scrapyd#302]
I've been trying for quite some time to use the listjobs endpoint without parameter, to retrieve running, pending, finished jobs but sadly the response contains null for project field:
```
khalistoo@ALTAIR:~$ curl http://localhost:6800/listjobs.json | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 17068  100 17068    0     0   132k      0 --:--:-- --:--:-- --:--:--  133k
{
    "finished": [
        {
            "end_time": "2018-12-24 04:14:05.667179",
            "id": "ed542114071e11e9b7e6000d3a2410d4",
            "project": null,
            "spider": "spider",
            "start_time": "2018-12-24 01:55:02.482214"
        },
        {
            "end_time": "2018-12-24 04:48:27.803269",
            "id": "ad9c6a6f073611e9b7e6000d3a2410d4",
            "project": null,
            "spider": "spider",
            "start_time": "2018-12-24 04:45:04.118479"
        },
        {
            "end_time": "2018-12-24 04:51:59.065684",
            "id": "b577ea5a072811e9b7e6000d3a2410d4",
            "project": null,
            "spider": "spider",
            "start_time": "2018-12-24 03:05:03.709788"
        }
    ],
    "node_name": "MOPS-EU-W-SCP5",
    "pending": [],
    "running": [
        {
            "id": "7537997e076c11e9b7e6000d3a2410d4",
            "pid": 12904,
            "project": null,
            "spider": "spider",
            "start_time": "2018-12-24 11:10:01.659171"
        },
        {
            "id": "7ea7e5a4077b11e9b7e6000d3a2410d4",
            "pid": 23648,
            "project": null,
            "spider": "spider",
            "start_time": "2018-12-24 12:57:40.188874"
        }
    ],
    "status": "ok"
}
```

Any idea of where I should look for. The webservice.py seems perfectly correct to me

**Comment by Digenis** [comment]
@khalistoo, @my8100,
The project names are the keys to the queues dictionary.
https://github.com/scrapy/scrapyd/blob/09722ea92db67520b7b80d45eea050f7e178d5b8/scrapyd/webservice.py#L121-L126
The above snippet needs to be rewritten.

**Comment by my8100** [comment]
@Digenis I addressed this issue in #318
